### PR TITLE
Pricing issue

### PR DESCRIPTION
**Pull request by fp-ps**

### CODE REVIEW

File: backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/model/validator/M_ProductPrice.java [hunk 1]
```diff
@@ -21,51 +35,103 @@
...
+
+			productPrice.setIsHUPrice(true);
 		}
-		else if (productPrice.getM_HU_PI_Item_Product().isInfiniteCapacity())
+		else
 		{
 			productPrice.setIsHUPrice(false);
 		}
+
+		validatePriceUOM(productPrice);
```
Comment (metas-ts) [inline backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/model/validator/M_ProductPrice.java:94]: i think if it's *not* a TU-UOM, then we shouldn't execute this check..right?

File: backend/de.metas.business/src/main/java/de/metas/pricing/attributebased/impl/AttributePricing.java [hunk 2]
```diff
@@ -129,9 +135,16 @@ protected void setResultForProductPriceAttribute(
 		final I_M_PriceList_Version pricelistVersion = loadOutOfTrx(productPrice.getM_PriceList_Version_ID(), I_M_PriceList_Version.class);
 		final I_M_PriceList priceList = pricelistVersion.getM_PriceList();
 
-		result.setPriceStd(productPrice.getPriceStd());
-		result.setPriceList(productPrice.getPriceList());
-		result.setPriceLimit(productPrice.getPriceLimit());
+		final ProductScalePriceService.ProductPriceSettings productPriceSettings = productScalePriceService.getProductPriceSettings(productPrice, pricingCtx.getQuantity());
+		if (productPriceSettings == null)
+		{
+			logger.trace("No ProductPriceSettings returned for qty : {} and M_ProductPrice_ID: {}", pricingCtx.getQty(), productPrice.getM_ProductPrice_ID());
```
Comment (metas-ts) [inline backend/de.metas.business/src/main/java/de/metas/pricing/attributebased/impl/AttributePricing.java:141]: please log to a Loggable.withLogger(logger, Level.DEBUG)

File: backend/de.metas.business/src/main/java/de/metas/pricing/rules/PriceListVersion.java [hunk 3]
```diff
@@ -73,16 +74,23 @@ public void calculate(@NonNull final IPricingContext pricingCtx, @NonNull final
 			return;
 		}
 
+		final ProductScalePriceService.ProductPriceSettings productPriceSettings = productScalePriceService.getProductPriceSettings(productPrice, pricingCtx.getQuantity());
+		if (productPriceSettings == null)
+		{
+			logger.trace("No ProductPriceSettings returned for qty : {} and M_ProductPrice_ID: {}", pricingCtx.getQty(), productPrice.getM_ProductPrice_ID());
```
Comment (metas-ts) [inline backend/de.metas.business/src/main/java/de/metas/pricing/rules/PriceListVersion.java:80]: please log to a Loggable.withLogger(logger, Level.DEBUG)

File: backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/model/validator/M_ProductPrice.java (hunk 1 above)
Comment (metas-ts) [reply backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/model/validator/M_ProductPrice.java]: changed this. please check if it's OK

Comment (fp-ps) [reply backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/model/validator/M_ProductPrice.java]: I'll test the change and come back with an answer

**Pull request merged**